src/directory.rs

```diff
@@ -199,52 +200,61 @@ impl Directory {
     }
 }
 
-impl Diff {
-    pub fn print(&self) {
-        fn go_added(path: &Path, entry: &DirectoryEntry) {
+impl fmt::Display for Diff {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
```

Comment:
Refactoring to use `Display` trait, now I seem to need to pass this `Formatter` around since it's not closed over in these internal functions. This was surprising to me, maybe reading up on how [closures work](https://doc.rust-lang.org/book/ch13-01-closures.html) could help or maybe refactoring all this to be non-recursive would avoid the need.

Also I think I need to figure out how to write tests for this 😓 these changes seem to break things. In a small repo I've set up for manual testing, before these changes
```
Code/rust/foo
❯ revtool diff dev
A foo/bar/baz/asdf.txt
```
and after
```
Code/rust/foo
❯ revtool diff dev
A foo1
```
(I had added a `foo1` directory, but the previous iteration didn't handle empty directories, but now we're showing that but missing the added file for some reason)

Reply:
> but now we're showing that but missing the added file for some reason

Oh, because of early `return` 🙃  (before we were just printing in each branch)